src/domain/core/src/services/query_service.rs

```diff
@@ -93,14 +93,6 @@ pub trait QueryService: Send + Sync {
         options: GetDataOptions,
     ) -> Result<GetDataResponse, QueryError>;
 
-    /// Extended version of [get_data()](Self::get_data()) that allows
-    /// retrieving requested metadata as well.
-    async fn get_data_with_metadata(
-        &self,
-        dataset_ref: &odf::DatasetRef,
-        options: GetDataWithMetadataOptions,
-    ) -> Result<GetDataWithMetadataResponse, QueryError>;
```

Comment:
> `get_data_with_metadata` looks like such a narrow specific method that gathers a bunch of loosely-related information ...
>
> I wonder if it would be better to have `QueryService::get_changelog_projection()` method that encapsulates everything that is currently going on in this UDTF.
>
> Seems like a much simpler API, less prone to misuse.
>
> This would also allow `QueryService` in future to speed up projections e.g. by caching them in a database somewhere - all this logic would stay hidden inside of the service.
(c) https://github.com/kamu-data/kamu-cli/pull/1428#discussion_r2479647940